Source/WebCore/Modules/filesystemaccess/WorkerFileSystemStorageConnection.cpp

```diff
@@ -274,7 +274,7 @@ void WorkerFileSystemStorageConnection::closeSyncAccessHandle(FileSystemHandleId
     if (!m_scope)
         return;
 
-    callOnMainThread([mainThreadConnection = m_mainThreadConnection, identifier, accessHandleIdentifier]() mutable {
+    callOnMainThreadAndWait([mainThreadConnection = m_mainThreadConnection, identifier, accessHandleIdentifier]() mutable {
         mainThreadConnection->closeSyncAccessHandle(identifier, accessHandleIdentifier);
```

Comment:
Can we avoid a sync IPC by using a BinarySemaphore in worker thread and an Async Reply in main thread, the reply triggering the semaphore signal?

Reply:
Sure, I will make the change.